From e96cffd3bcdf144921632e7883e981e7a75a1ac5 Mon Sep 17 00:00:00 2001
From: Vincent Bernat <vincent@bernat.im>
Date: Sat, 17 Dec 2016 23:39:10 +0100
Subject: [PATCH] * uacctd: use current time when we don't have a timestamp
 from netlink

---
 src/uacctd.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/uacctd.c b/src/uacctd.c
index 363acb3082bf..e0f068d1e2ef 100644
--- a/src/uacctd.c
+++ b/src/uacctd.c
@@ -48,7 +48,6 @@ static int nflog_incoming(struct nflog_g_handle *gh, struct nfgenmsg *nfmsg,
                           struct nflog_data *nfa, void *p)
 {
   static char jumbo_container[10000];
-  struct timeval tv = {};
   struct pcap_pkthdr hdr;
   char *pkt = NULL;
   ssize_t pkt_len = nflog_get_payload(nfa, &pkt);
@@ -66,8 +65,9 @@ static int nflog_incoming(struct nflog_g_handle *gh, struct nfgenmsg *nfmsg,
 
   if (pkt_len == -1)
     return -1;
-  nflog_get_timestamp(nfa, &tv);
-  hdr.ts = tv;
+  if (nflog_get_timestamp(nfa, &hdr.ts) < 0) {
+    gettimeofday(&hdr.ts, NULL);
+  }
   hdr.caplen = MIN(pkt_len, config.snaplen);
   hdr.len = pkt_len;
 
-- 
2.11.0

